forum/search/typesense.py

```diff
@@ -0,0 +1,423 @@
+"""
+Typesense backend for searching comments and threads.
+"""
+
+from typing import Any, Optional
+
+from bs4 import BeautifulSoup
+from django.conf import settings
+from django.core.paginator import Paginator
+
+# mypy complains: 'Module "typesense" does not explicitly export attribute "Client"'
+from typesense import Client  # type: ignore
```

Comment:
I have no idea why mypy is complaining about this. I thought it might be confused since the current module is also `typesense`, but that doesn't seem to be the case. Anyhow, you can use `from typesense.client import Client` if you want, which should then give `Client` proper typing throughout this file.